testutil/network/basenet/accounts.go

```diff
@@ -0,0 +1,194 @@
+		memnet.Config.Keyring = keyring.NewInMemory(memnet.Config.CosmosCfg.Codec)
+	} else {
+		t.Log("Keyring already initialized, using existing keyring")
+	}
+
+	// Create memnet.NumKeyringAccounts() number of accounts in the configured keyring.
+	accts := testkeyring.CreatePreGeneratedKeyringAccounts(
+		t, memnet.Config.Keyring, memnet.Config.GetNumKeyringAccounts(t),
+	)
+
+	// Assign the memnet's pre-generated accounts to a new pre-generated
+	// accounts iterator containing only the accounts which were also created
+	// in the keyring.
+	memnet.PreGeneratedAccountIterator = testkeyring.NewPreGeneratedAccountIterator(accts...)
+}
+
+// CreateOnChainAccounts creates on-chain accounts (i.e. auth module) for the sum of
+// the configured number of suppliers, applications, and gateways.
+func (memnet *BaseInMemoryNetwork) CreateOnChainAccounts(t *testing.T) {
+	t.Helper()
+
+	// NB: while it may initially seem like the memnet#Init<actor>AccountsWithSequence() methods
+	// can be refactored into a generic function, this is not possible so long as the genesis
+	// state lists are passed directly & remain a slice of concrete types (as opposed to pointers).
+	// Under these conditions, a generic function would not be able to unmarshal the genesis state
+	// stored in the in-memory network because it is unmarshalling uses reflection, and it is not
+	// possible to reflect over a nil generic type value.
+
+	// Retrieve the supplier module's genesis state from cosmos-sdk in-memory network.
+	supplierGenesisState := network.GetGenesisState[*suppliertypes.GenesisState](t, suppliertypes.ModuleName, memnet)
+	if supplierGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "supplier")
+	} else {
+		// Initialize on-chain accounts for genesis suppliers.
+		memnet.InitSupplierAccountsWithSequence(t, supplierGenesisState.SupplierList...)
+
+	}
+
+	// Retrieve the application module's genesis state from cosmos-sdk in-memory network.
+	appGenesisState := network.GetGenesisState[*apptypes.GenesisState](t, apptypes.ModuleName, memnet)
+	if appGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "application")
+	} else {
+		// Initialize on-chain accounts for genesis applications.
+		memnet.InitAppAccountsWithSequence(t, appGenesisState.ApplicationList...)
+	}
+
+	// Retrieve the gateway module's genesis state from cosmos-sdk in-memory network.
+	gatewayGenesisState := network.GetGenesisState[*gatewaytypes.GenesisState](t, gatewaytypes.ModuleName, memnet)
+	if gatewayGenesisState == nil {
+		t.Logf(warnNoModuleGenesisFmt, "gateway")
+	} else {
+		// Initialize on-chain accounts for genesis gateways.
+		memnet.InitGatewayAccountsWithSequence(t, gatewayGenesisState.GatewayList...)
+	}
+
+	// need to wait for the account to be initialized in the next block
+	require.NoError(t, memnet.GetNetwork(t).WaitForNextBlock())
+}
+
+// InitAccount initializes an Account by sending it some funds from the validator
+// in the network to the address provided
```

Comment:
Could you expand on this comment as to **why** we need to initialise an Account (pubkey, give it some funds etc)